### ISSUE TITLE
Tiktok is detecting puppeteer-extra-plugin-stealth

### ISSUE THREAD

**Issue by MerwinV1** [berstend/puppeteer-extra#218]
Hi everyone,

I'm curently working on a tiktok scraper, using the stealth plugin from the begining of it, but today when i'm trying to go on a tiktok profile i'm getting this:

<img width="1195" alt="Capture d’écran 2020-06-05 à 13 00 01" src="https://user-images.githubusercontent.com/26410335/83871606-a19e5b00-a730-11ea-87d3-799403c01841.png">

Any Idea how to fix it ? 

For the code i'm using (code for testing the connection):

```
const puppeteer = require('puppeteer-extra');
const StealthPlugin = require('puppeteer-extra-plugin-stealth')
var fs = require("fs");

(async () => {
    try {
      puppeteer.use(StealthPlugin())
      var browser = await puppeteer.launch({
          headless: false,
        });
      var page = await browser.newPage();
      await page.goto(`https://www.tiktok.com/@miss_extra2.0`);
      await page.waitFor(50000)
      await browser.close();
    } catch (err) {
      await browser.close();
    }
  })();

```

**Comment by danielgwilson** [comment]
Also experiencing this issue

**Comment by Torpedo99** [comment]
Google Chrome seems to be detected by sites lately. Using puppeteer-firefox is a good way to scrape Tiktok. Hoping someone could find out why chrome lets sites know about automation as this is only a temporary fix.

**Comment by danielgwilson** [comment]
@Torpedo99 thanks for that—trying now

**Comment by delmotte** [comment]
up

It works with puppeteer-firefox but it would be better to see it working with puppeteer/puppeteer-extra and stealth plugin

**Comment by Daniel-Griffiths** [comment]
Same here, had a lot of issues with puppeteer being detected recently. The firefox work around works for now :slightly_smiling_face:

**Comment by rezemble** [comment]
it's already the `--remote-debugging-port` flag that trips the detection.

running chromium (bare) without loads fine,
once the debugging port is set, the profile request doesn't return anything.

**Comment by Daniel-Griffiths** [comment]
@rezemble How did you disable the debugging port?

I am guessing it’s just another launch option you set to false?

**Comment by rezemble** [comment]
@Daniel-Griffiths well I took the chromium binary provided with puppeteer and ran it without any flags, opened a page, loaded fine
then started the same binary with `--remote-debugging-port=0`, opened a page, doesn't load.
without puppeteer.

So it would seem they're somehow detecting that remote debugging is enabled

**Comment by rezemble** [comment]
you can't use puppeteer without the remote debugging port, it's what allows it to interact with the browser.

**Comment by Daniel-Griffiths** [comment]
Thanks for the explination @rezemble I guess that's going to be a tricky one to work around for now. 

Sites keep geting better at dected pupeteer :sweat_smile:

**Comment by vityas-off** [comment]
Somehow i ended with this
```js
const puppeteer = require("puppeteer-extra");
const stealthPlugin = require("puppeteer-extra-plugin-stealth")();

["chrome.runtime", "navigator.languages"].forEach(a =>
  stealthPlugin.enabledEvasions.delete(a)
);

puppeteer.use(stealthPlugin);

main();
async function main() {
  const browser = await puppeteer.launch();
  const page = await browser.newPage();

  await page.evaluateOnNewDocument(() => {
    delete navigator.__proto__.webdriver;
  });

  await page.goto(`https://www.tiktok.com/@miss_extra2.0`);
  await page.screenshot({ path: "./11.png" });
  browser.close();
}
```

**Comment by momala454** [comment]
https://blog.chromium.org/2018/06/improving-extension-transparency-for.html

`window.chrome.webstore` is undefined now, it should be removed from here https://github.com/berstend/puppeteer-extra/blob/master/packages/puppeteer-extra-plugin-stealth/evasions/shared.js

but it should be also improved further, you can detect the fake functions with
`window.chrome.csi.toString()`

**Comment by olitomas** [comment]
Related to this? 🤔
https://github.com/digitalhurricane-io/puppeteer-detection-100-percent

**Comment by momala454** [comment]
tiktok calls 
`chrome.runtime.connect`

removing this function make tiktok works, but one should probably simulate its functionnality to prevent being detected using this

**Comment by berstend** [comment]
This should be fixed as soon as #292 is merged and published. :-)

**Comment by berstend** [comment]
Published in `puppeteer-extra-plugin-stealth@2.6.0`
